src/MvvmNano.Core/MvvmNanoViewModel.cs

```diff
@@ -36,6 +36,19 @@ protected void NotifyPropertyChanged([CallerMemberName]string propertyName = nul
             PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
         }
 
+        /// <summary>
+        /// Sets the property and notifies listeners only when necessary.
+        /// </summary>
+        /// <typeparam name="T">Type of the property.</typeparam>
+        /// <param name="storage">Reference to a property with both getter and setter.</param>
+        /// <param name="value">Value for the property.</param>
+        /// <param name="propertyName">Name of the property used to notify listeners. This value is optional and can be provided automatically when invoked from compilers that support CallerMemberName.</param>
+        protected void SetProperty<T>(ref T storage, T value, [CallerMemberName] string propertyName = null)
+        {
+            storage = value;
```

Comment:
Should we add an equality check here to avoid unnecessary PropertyChanged events?